src/Mindee/Extraction/ImageExtractor.cs

```diff
@@ -0,0 +1,319 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Docnet.Core;
+using Docnet.Core.Models;
+using Mindee.Exceptions;
+using Mindee.Geometry;
+using Mindee.Input;
+using Mindee.Parsing.Standard;
+using SkiaSharp;
+
+namespace Mindee.Extraction
+{
+    /// <summary>
+    /// Extract sub-images from an image.
+    /// </summary>
+    public class ImageExtractor
+    {
+        private readonly List<SKBitmap> _pageImages;
+        private readonly string _filename;
+        private readonly string _saveFormat;
+
+        /// <summary>
+        /// Init from a Local Input Source.
+        /// </summary>
+        /// <param name="localInput"></param>
+        public ImageExtractor(LocalInputSource localInput)
+        {
+            this._filename = localInput.Filename;
```

Comment:
no need to use `this`

Reply:
sweet

Reply:
Need to go through the code, there are many other instances of this.